HW20/Deck.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <vector>
+#include<string>
+
+template<typename T>
+
+class Deck
+{
+public:
+	Deck() {};
+	Deck(const std::vector<T>& initDataBase) : m_initDataBase(initDataBase), m_availableDataBase(initDataBase) {};
+	~Deck() {};
+
+	T getRandomCard()
+	{
+		if (m_availableDataBase.empty())
+		{
+			resetDatabase();
+		}
+		int choice = std::rand() % m_availableDataBase.size();
+		T monster = m_availableDataBase[choice];
+		m_availableDataBase.erase(m_availableDataBase.begin() + choice);
```

Comment:
std::remove would be enough here and it would be more efficient for this case as you don't care about internal structure of elements